lib/money/money.rb

```diff
@@ -619,6 +658,23 @@ def decimal_mark
 
   private
 
+  def highest_precision(other)
+    infinite_precision? || other.infinite_precision?
+  end
+
+  def dup_with(options = {})
+    fractional ||= options[:fractional] || self.fractional
+    currency ||= options[:currency] || self.currency
+    bank ||= options[:bank] || self.bank
+    infinite_precision =
+      if options[:infinite_precision].nil?
+        self.infinite_precision?
+      else
+        options[:infinite_precision]
+      end
+    self.class.new(fractional, currency, { bank: bank, infinite_precision: infinite_precision })
+  end
```

Comment:
It could also be done like this:
```ruby
  def dup_with(options = {})
    self.class.new(
      options[:fractional] || fractional,
      options[:currency] || currency,
      bank: options[:bank] || bank,
      infinite_precision:
        options.fetch(:infinite_precision, infinite_precision?)
    )
  end
```
The difference here is that we do not define extra variables, and we do not check if `options[:infinite_precision].nil?`, but we check if that key was present in `options` instead.

Reply:
I like this stylistic suggestion 👍